codespan/src/span.rs

```diff
@@ -9,7 +10,7 @@ pub struct Span<I> {
     end: I,
 }
 
-impl<I: Index> Span<I> {
+impl<I: Ord> Span<I> {
```

Comment:
Why this change?

Reply:
I guess I was trying to limit it to fixed locations, rather than potentially folks using it for offsets. But I can see you reasoning, so let's go with that.

Reply:
Okay, I figured it was just out of convenience that all functions were bounded by `I: Index`.  I could implement `Index` for my `Location` type and restore this bound so that guarantee remains instead (though `start` and `end` ought to still stay unbounded)

Reply:
Yeah, have a look - it might be a pain to implement for `Location` - eg. you might need to then introduce a corresponding `LocationOffset` type. I understand if that's a bit much to ask for though!